src/org/usfirst/frc4904/robot/commands/AutoShifter.java

```diff
@@ -0,0 +1,68 @@
+package org.usfirst.frc4904.robot.commands;
+
+
+import java.util.LinkedList;
+import org.usfirst.frc4904.robot.RobotMap;
+import org.usfirst.frc4904.standard.custom.sensors.CustomEncoder;
+import org.usfirst.frc4904.standard.subsystems.chassis.AutoSolenoidShifters;
+import org.usfirst.frc4904.standard.subsystems.chassis.SolenoidShifters;
+import edu.wpi.first.wpilibj.command.Command;
+
+public class AutoShifter extends Command {
+	protected CustomEncoder leftEncoder;
+	protected CustomEncoder rightEncoder;
+	protected AutoSolenoidShifters shifter;
+	protected LinkedList<Double> speeds = new LinkedList<Double>();
+	public static final double RAPID_ACCELERATION = 8;// WIP
+	public static final double RAPID_DECELERATION = 2;// WIP
+	public static final double RATE_DIFF = 100; // encoder ticks--WIP
+	public static final double SUPER_SLOW_RATE = 2; // ft per sec
+	public static final double MEDIUM_RATE = AutoShifter.SUPER_SLOW_RATE * 2;
+	public static final double FAST_RATE = AutoShifter.SUPER_SLOW_RATE * 4;
+	public static final double SLOW_THROTTLE = 0.4;
+	public static final double MEDIUM_THROTTLE = AutoShifter.SLOW_THROTTLE * 1.875;
+	public static final double FAST_THROTTLE = AutoShifter.SLOW_THROTTLE * 2;
+	public static final double SPEED_LIST_MAX_LENGTH = 5;
+	public static final double LAST_MANUAL_SHIFT_TIME = 5000; // TODO
+	public static final double LAST_AUTO_SHIFT_TIME = 500;// TODO
+
```

Comment:
Should have a default constructor that calls `this()` with references to the components in RobotMap.